src/Components/UI/Navigation/DesktopNavigation/VerticalNavBar.jsx

```diff
@@ -0,0 +1,64 @@
+import { Link, useLocation } from "react-router-dom";
+import { useTheme } from "../../../../Context";
+import IconButton from "../../Button/IconButton";
+import "../VerticalNavBar.css";
+
+export const VerticalNavBar = () => {
+  const { theme, toggleTheme } = useTheme();
+  const { pathname } = useLocation();
+
+  const activeHome = pathname.includes("videos") ? "active-nav " : "";
+  const activeTrending = pathname.includes("trending") ? "active-nav " : "";
+  const activePlaylist = pathname.includes("playlist") ? "active-nav " : " ";
+  const activeWatchlater = pathname.includes("watchlater") ? "active-nav " : "";
+  const activeHistory = pathname.includes("history") ? "active-nav " : "";
+
+  const themeIcon = theme === "dark" ? "fa fa-sun" : "fa fa-moon";
+
+  return (
+    <div className="vertical-nav-container nav-bar-desktop">
+      <div className="vertical-nav-menu">
+        <h1>Menu</h1>
+        <Link to="/videos">
```

Comment:
instead of using the link you can use the `NavLink` for the styling by passing a callback func.

`<NavLink
            style={({ isActive }) => isActive ? "active-nav":""}
`

Reply:
Inside NavLink it works the same way, so I created a custom version of mine.  :)